src/dynamo.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+
+/*
+Table Schema = {
+	channel: 'C1H9RESGL',				// channel name from Slack API
+	data: '2019-11-01',					// date entered by user
+	slackId: '1503435956.000247,		// ts value from Slack API
+	questionText: 'some text',			// text entered by user, may be zero length
+	answer: 'fact',						// 'fact' or 'crap'
+	answerText: 'some text',			// text entered by user, may be zero length
+	users: {							// map with one key per user
+		UB9NQLNNR: {					// username from Slack API
+			timestamp: 1573618530557,	// Unix time of first vote in milliseconds
+			answer: 'fact'				// 'fact' or 'crap'
+		},
+		...
+	}
+}
+*/
+
+const TABLE_NAME = process.env.TABLE_NAME;
+
+const marshall = AWS.DynamoDB.Converter.marshall;
+const dynamodb = new AWS.DynamoDB({region: process.env.AWS_REGION});
+
+async function updateItem(params) {
+	try {
+		await dynamodb.updateItem(params).promise();
+		return true;
+	} catch (err) {
+		if (err.code === 'ConditionalCheckFailedException') {
+			return false;
+		}
+		throw err;
+	}
+}
+
+module.exports.create = async ({channel, date, questionText}) => {
+	return await updateItem({
+		TableName: TABLE_NAME,
+		Key: marshall({channel, date}),
+		ExpressionAttributeNames: {
+			'#c': 'channel',
+			'#qt': 'questionText'
+		},
+		ExpressionAttributeValues: marshall({':qt': questionText}),
+		UpdateExpression: 'SET #qt = :qt',
+		ConditionExpression: 'attribute_not_exists(#c)'
+	});
+};
+
+module.exports.close = async ({channel, date, answer, answerText}) => {
+	return await updateItem({
+		TableName: TABLE_NAME,
+		Key: marshall({channel, date}),
+		ExpressionAttributeNames: {
+			'#c': 'channel',
+			'#a': 'answer',
+			'#at': 'answerText'
+		},
+		ExpressionAttributeValues: marshall({':a': answer, ':at': answerText}),
+		UpdateExpression: 'SET #a = :a, #at = :at',
+		ConditionExpression: 'attribute_exists(#c) and attribute_not_exists(#a)'
+	});
+};
+
+module.exports.setSlackId = async ({channel, date, slackId}) => {
```

Comment:
Called by the stream lambda once it has posted the initial question message. The `slackId` parameter is the `ts` parameter in the slack API response. We need to save this as it's required to update an existing message. The stream lambda will have to handle that writing this value back to the table will invoke the stream lambda again.